---
title: "Adaptation of the rumen microbiota during a finishing study"
author: "Christopher L. Anderson (canderson30@unl.edu)"
date: ""
output: html_document
    keep_md: yes
---

# Adaptation of the rumen microbiota during a finishing study
Christopher L. Anderson, Samodha C. Fernando

## Introduction
This is a R Markdown file to accompany the mansucript titled, "Adaptation of the rumen microbiota during a finishing study." It was written in [R markdown](http://rmarkdown.rstudio.com) and converted to html using the R knitr package. This enables us to embed the results of our analyses directly into the text to allow for a reproducible data analysis pipeline. A [github repository is available](https://github.com/chrisLanderson/rumen_adaptation). All of the commands below have been ran in linux environment (did not have root privileges).  Should work on Mac as well if you install a Mac version of R and USEARCH.


Linux version used on UNL HPC to generate manuscript outputs:
cat /proc/version
Linux version 2.6.32-504.23.4.el6.x86_64 (mockbuild@sl6.fnal.gov) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC) ) #1 SMP Tue Jun 9 11:55:03 CDT 2015


## **BEFORE YOU RENDER WITH KNITR**: To recreate the anlaysis used in the Anderson et al. manuscript,  there are two steps (follow the guidelines below). All of the commands to generate the manuscript outpurs have been ran in linux enviornment (did not have root privileges).  Should work on Mac as well if you install a Mac version of R and USEARCH. Only dependency is perl, which comes on all Mac and linux systems and to my knowledge, the version shouldn't matter for running the few custom scripts.

  1. Run the bash script to create a virtual enironment and download/install the following programs **LOCALLY** so everything is the same version used in the manuscript:
    + Python
    + QIIME
    + FASTX
    + Mothur
    + USEARCH
    + R Markdown file to render

  2. Render the R Markdown file with knitR to recreate the workflow and outputs.

Due to licensing issues, USEARCH can not be included in the setup. To obtain a download link, go to the USEARCH [download page](http://www.drive5.com/usearch/download.html) and select version USEARCH v7.0.1090 for linux. **A link (expires after 30 days) will be sent to the provided email. Use the link as an argument for shell script below**.

Simply download the bash script from the github repository and run it (**provide the link to download your licensed USEARCH version as an argument for setup.sh**):

  1. wget https://raw.githubusercontent.com/chrisLanderson/rumen_adaptation/master/setup.sh
  2. chmod 775 setup.sh 
  3. ./setup.sh usearch_link


**Note: It will take some time to install everything**


To start a R session, run these two commands from within the rumenEnv/ direcotry:

  1. source bin/activate
  2. R

To convert the R markdown to html (or any other format) use the [knitr package](http://yihui.name/knitr/) from within R using the command: **knit2html("rumen_adaptation.Rmd")**.

The following packages and knitR settings were used to compile this document:

```{r, warning=FALSE}
#cat("library(knitr)", "library(ggplot2)", "library(vegan)", #"library(matrixStats)", "library(grid)", "library(reshape)", #"library(biom)", "library(gplots)", "library(Heatplus)", #"library(RColorBrewer)",
 #   sep="\n", 
#    file="checkpoint_code.R")

#install.packages("checkpoint")
#library(checkpoint)
#checkpoint("2015-06-30",  checkpointLocation = getwd())

# deps = c("knitr", "ggplot2", "vegan", "matrixStats", "grid", "reshape", "gplots", "biom", "RColorBrewer");
# 
# for (dep in deps){
#   if (dep %in% installed.packages()[,"Package"] == FALSE){
#     install.packages(as.character(dep), quiet=TRUE,
#                      repos='http://cran.us.r-project.org');
#   }
#   library(dep, verbose=FALSE, character.only=TRUE)
# }

#install HeatPlus from bioconductor

sessionInfo()

perm = 1e4
```

```{r knitr.settings, eval=TRUE, echo=TRUE, cache=TRUE}
opts_chunk$set("fig.path"="figure/")
opts_chunk$set("fig.align"="center")
opts_chunk$set("dev" = c("png", "pdf"))

opts_chunk$set("tidy" = TRUE)

opts_chunk$set("echo" = TRUE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = TRUE)
```

## Data curation

Download the 454 data (not really raw data though because the sequencing center had done some initial quality control on them...see manuscript for details):
```{r, engine='bash', results='hide'}
#wget 129.93.221.145:/public/rumen.adaptation.fasta
```

Now  get the SILVA reference alignment and trim it to the V1-V3 region (region was determined using 27F and 518R primers) of the 16S rRNA gene.
```{r, engine='bash', results='hide'}
# wget http://www.mothur.org/w/images/2/27/Silva.nr_v119.tgz
# tar -zxvf Silva.nr_v119.tgz
# rm -rf Silva.nr_v119.tgz __MACOSX silva.silva.nr_v119.tax README.Rmd README.html README.md
```

## Demulitplex and Quality Control

The code chunk below demulitplexes the seqeuncing library using the provided mapping file from github then trims off the reverse primer.  Subseqeuntly, we use a combination of mothur and FASTX to trim the seqeunces to a fixed length of 400 basepairs to improve OTU picking in UPARSE downstream. Finally, the sequences are reverse complemented in mothur.

```{r, engine='bash', results='hide'}
#  wget https://raw.githubusercontent.com/chrisLanderson/rumen_adaptation/master/mapping.txt
#  
#  wget https://raw.githubusercontent.com/chrisLanderson/rumen_adaptation/master/qiime_parameters_working.txt
#  
#  split_libraries.py -m mapping.txt -f rumen.adaptation.fasta -b hamming_8 -l 0 -L 1000 -M 1 -o rumen_adaptation_demultiplex
#  
#  truncate_reverse_primer.py -f rumen_adaptation_demultiplex/seqs.fna -o rumen_adaptation_rev_primer -m mapping.txt -z truncate_only -M 2
#  
#  
#  mothur "#trim.seqs(fasta=rumen_adaptation_rev_primer/seqs_rev_primer_truncated.fna, minlength=400)"
#  
#  fastx_trimmer -i rumen_adaptation_rev_primer/seqs_rev_primer_truncated.trim.fasta -l 400 -o rumen.adaptation.qc.trim.fasta
 
mothur "#reverse.seqs(fasta=rumen.adaptation.qc.trim.fasta)"
```

Use a custom perl script from our lab to convert the fasta file from QIIME format to a format that works with UPARSE to generate the OTU table:

```{r, engine='bash', results='hide'}
wget https://raw.githubusercontent.com/chrisLanderson/rumen_adaptation/master/qiime_to_usearch.pl
chmod 775 qiime_to_usearch.pl
./qiime_to_usearch.pl -fasta=rumen.adaptation.qc.trim.rc.fasta -prefix=rumen
mv format.fasta rumen.adaptation.format.fasta
```

Run the seqeunces through the UPARSE pipeline to pick OTUs:
```{r, engine='bash'}

svn export https://github.com/chrisLanderson/rumen_adaptation/trunk/usearch_python_scripts

chmod -R 775 usearch_python_scripts

wget https://github.com/chrisLanderson/rumen_adaptation/raw/master/gold.fasta.gz
chmod 775 uc2otutab.py
chmod 775 fasta_number.py
gzip -d gold.fasta.gz
chmod 775 gold.fasta


mkdir usearch_results

usearch -derep_fulllength rumen.adaptation.format.fasta -sizeout -output usearch_results/rumen.adaptation.derep.fasta

usearch -sortbysize usearch_results/rumen.adaptation.derep.fasta -minsize 2 -output usearch_results/rumen.adaptation.derep.sort.fasta

usearch -cluster_otus usearch_results/rumen.adaptation.derep.sort.fasta -otus usearch_results/rumen.adaptation.otus1.fasta

usearch -uchime_ref usearch_results/rumen.adaptation.otus1.fasta -db gold.fasta -strand plus -nonchimeras usearch_results/rumen.adaptation.otus1.nonchimera.fasta

usearch_python_scripts/fasta_number.py usearch_results/rumen.adaptation.otus1.nonchimera.fasta > usearch_results/rumen.adaptation.otus2.fasta

usearch -usearch_global rumen.adaptation.format.fasta -db usearch_results/rumen.adaptation.otus2.fasta -strand plus -id 0.97 -uc usearch_results/rumen.adaptation.otu_map.uc

usearch_python_scripts/uc2otutab.py usearch_results/rumen.adaptation.otu_map.uc > usearch_results/rumen.adaptation.otu_table.txt

cp usearch_results/rumen.adaptation.otu_table.txt ./

```

Assign taxonomy to the OTU representative sequences.  Can't find the RDP jar file in the QIIME installation, if it was even there...so download it remotely:
```{r, engine='bash'}
assign_taxonomy.py -i usearch_results/rumen.adaptation.otus2.fasta -t lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/taxonomy/97_otu_taxonomy.txt -r lib/python2.7/site-packages/qiime_default_reference/gg_13_8_otus/rep_set/97_otus.fasta -o assign_gg_taxa -m mothur

```
