---
title: "Adaptation of the rumen microbiota during a finishing study"
author: "Christopher L. Anderson"
date: ""
output: html_document
    keep_md: yes
---

#Adaptation of the rumen microbiota during a finishing study
Christopher L. Anderson, Samodha C. Fernando

##Introduction
This is a R Markdown file to accompany the mansucript titled, "Adaptation of the rumen microbiota during a finishing study." It was written in [R markdown](http://rmarkdown.rstudio.com) and converted to html using the R knitr package. This enables us to embed the results of our analyses directly into the text to allow for a reproducible data analysis pipeline. A [github repository is available](https://github.com/chrisLanderson/rumen_adaptation). All of the commands below have been ran in Mac OS X (version 10.9.5 but can use ealier). If you install a linux version of USEARCH and R instead, the R Markdown will (or should) render in a linux enviornment as well.

cat /proc/version
Linux version 2.6.32-504.23.4.el6.x86_64 (mockbuild@sl6.fnal.gov) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-11) (GCC) ) #1 SMP Tue Jun 9 11:55:03 CDT 2015

##**BEFORE BEGINNING**: To recreate the anlaysis used in the Anderson et al. manuscript,  there are two steps (follow the guidelines below, other dependendies assumed are python (used version 2.7.5) and perl (used v5.16.3):

 1. Run a bash script to create a virtual enironment, download USEARCH, download R, download and install QIIME, and download the R Markdown file to render.
  2. Render the R Markdown file with knitR to recreate the workflow and outputs.

Due to licensing issues, USEARCH can not be included or downloaded from a provided link. To obtain a download link, go to the USEARCH [download page](http://www.drive5.com/usearch/download.html) and select version USEARCH v7.0.1090 for Mac. **A link (expires after 30 days) will be sent to the provided email. The setup.sh script downloaded below will prompt you for the download link**.

Simply download the bash script from the github repository and run it (**the script will prompt you for the link to download USEARCH**):


**wget <shell script from github link>**

**bash script**


To convert the R markdown to html (or any other format) use the [knitr package](http://yihui.name/knitr/) from within R using the command: **knit2html("rumen_adaptation.Rmd")**.

The html rendered version can be found [here]()

The following packages and knitR settings were used to compile this document:

```{r, warning=FALSE}

packages = c("knitr", "ggplot2", "vegan", "matrixStats", "grid", "reshape", "biom", "gplots", "Heatplus", "RColorBrewer", "checkpoint");

for (package in packages){
  if (package %in% installed.packages()[,"Package"] == FALSE){
    install.packages(as.character(package), quiet=TRUE);
  }
  library(package, verbose=FALSE, character.only=TRUE)
}

sessionInfo()

perm = 1e4
```

```{r knitr.settings, eval=TRUE, echo=TRUE, cache=TRUE}
opts_chunk$set("fig.path"="figures/")
opts_chunk$set("fig.align"="center")
opts_chunk$set("dev" = c("png", "pdf"))

opts_chunk$set("tidy" = TRUE)

opts_chunk$set("echo" = TRUE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = TRUE)
```

##Download and install other bioinformatic tools

```{r, engine='bash', results='hide'}
wget 




```

## Data curation

Download the 454 data (not really raw data though because the sequencing center had done some initial quality control on them...see manuscript for details):

```{r, engine='bash', results='hide'}

```

Now  get the SILVA reference alignment and trim it to the V3 region of the 16S rRNA gene.
```{r, engine='bash', results='hide'}
wget http://www.mothur.org/w/images/9/98/Silva.bacteria.zip
unzip *zip
mv silva.bacteria/silva.bacteria.fasta ./
rm -rf silva.bacteria __MACOSX Silva.bacteria.zip
```
